entrypoint.sh

```diff
@@ -36,9 +36,25 @@ fi
 if [ -f /var/run/secrets/kubernetes.io/serviceaccount/token ]
 then
   KUBE_TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
-  export VAULT_TOKEN=$(curl --request POST \
-    --data '{"jwt": "'"$KUBE_TOKEN"'", "role": "'"$SERVICE_NAME"'"}' \
-    $VAULT_ADDR/v1/auth/kubernetes/login | jq -r '.auth.client_token')
+  vault_token="null"
+  attempts=1
+  while [ "${attempts}" -le 10 ]; do
+    echo "Attempt number ${attempts} to get vault token..."
+    vault_token=$(curl --request POST \
```

Comment:
We could use curl --fail | (increment attempts && sleep), but i'm not sure if that makes it any clearer.

Reply:
I'm not sure that it would be better. The way you did it looks fine and later we can add reporting to the bash script if we want.